### DISCUSSION TITLE
[5.x]: Some unexpected behaviour when clicking and selecting element table rows

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#14379] [Ideas] [answered]
### What happened?

In element indexes with table view:

1. **When clicking anywhere on a table row**, the element is selected/unselected. That's kind of unexpected, but maybe good for accessibility since the click area to select an element is so large.
2.  **When clicking on a row and then doing a `Ctrl + click` on another element**, the first element becomes deselected, and the second one is selected. _This definitely feels weird and gets me every single time_! The expectation would be for both elements to be selected. It may (?) be this way because you are somehow checking a checkbox, but I think most users are thinking of selecting elements like in a classic file explorer UI. This part feels to me like a bug or oversight, and is the reason I'm not posting this as a discussion. I may be missing something, so feel free to correct me if there is good reason (accessibility perhaps?) for this.
3. **Double clicking a row** cycles the element between selected/deselected states. This is kind of expected, because of no1, but feels like a wasted oppurtunity to _open the slideout_. As a matter of fact, this is what happens when the index is switched to `card view`. IMO this (the opening of the slideout via double-clicking on the row) would be very convenient and largely solve the problem of slideout discoverability. I have seen the `Improved checkbox-style deselection behavior for control panel items, to account for double-clicks.` item on the [beta changelog](https://github.com/craftcms/cms/blob/5.0/CHANGELOG.md#L22), not sure if it's related. I _think_ it worked like I'm suggesting at some point in the alpha?

### Craft CMS version

Craft CMS 5.0.0-beta.1

### Installed plugins and versions

-

**Comment by mmikkel** [comment]
Missed this thread during the beta, and came here via https://github.com/craftcms/cms/issues/14979.

I think the fact that the entire row is selectable is a good change, but agree on your second and third points. Especially the second, and especially considering that Craft 4 does have Finder/Explorer-like behavior for the Cmd/Ctrl key, albeit only in element select modals.

**Comment by brandonkelly** [comment, marked as answer]
> 1. **When clicking anywhere on a table row**, the element is selected/unselected.

That’s intended.

> 2\. **When clicking on a row and then doing a `Ctrl + click` on another element**, the first element becomes deselected, and the second one is selected.

Also intended. This is how element indexes (and Matrix blocks) have always worked in Craft. Pressing <kbd>Command</kbd>/<kbd>Ctrl</kbd> will cause the selection to mimic non-checkbox-based selects, in the same way that pressing <kbd>Command</kbd>/<kbd>Ctrl</kbd> on non-checkbox-based selects will cause them to mimic checkbox-based selects.

When the items you are selecting have checkboxes (as within element indexes) and you want to select multiple items, there is no need to press <kbd>Command</kbd>/<kbd>Ctrl</kbd> unless you want that alternate behavior.

> 3\. **Double clicking a row** cycles the element between selected/deselected states. This is kind of expected, because of no1, but feels like a wasted oppurtunity to _open the slideout_.

Agree, that makes sense, and makes even more sense coupled with #14968, where entries without thumbnails effectively wouldn’t have a direct hit area for double-clicking anymore, when the Status column is shown. So just made that change for 5.2 (43fb8d854b6fbfafde7401f3dcfa02426cffa92c).

**Reply by brandonkelly** [reply]
Craft 5.2.0-beta.1 is out with that change.
